cpp/examples/low_level_reader_example.cc

```diff
@@ -0,0 +1,153 @@
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+#include <iostream>
+
+#include "arrow/api.h"
+#include "arrow/filesystem/api.h"
+
+#include "./config.h"
+#include "gar/reader/chunk_info_reader.h"
+
+void vertex_property_chunk_info_reader(
+    const GAR_NAMESPACE::GraphInfo& graph_info) {
+  // constuct reader
+  std::string label = "person", property_name = "id";
+  ASSERT(graph_info.GetVertexInfo(label).status().ok());
+  auto maybe_group = graph_info.GetVertexPropertyGroup(label, property_name);
+  ASSERT(!maybe_group.has_error());
+  const GAR_NAMESPACE::PropertyGroup& group = maybe_group.value();
+  auto maybe_reader = GAR_NAMESPACE::ConstructVertexPropertyChunkInfoReader(
+      graph_info, label, group);
+  ASSERT(!maybe_reader.has_error());
+  GAR_NAMESPACE::VertexPropertyChunkInfoReader& reader = maybe_reader.value();
+
+  // use reader
+  auto maybe_chunk_path = reader.GetChunk();
+  ASSERT(maybe_chunk_path.status().ok());
+  std::string chunk_path = maybe_chunk_path.value();
+  std::cout << "path of first vertex property chunk: " << chunk_path
+            << std::endl;
+  // seek vertex id
+  ASSERT(reader.seek(520).ok());
+  maybe_chunk_path = reader.GetChunk();
+  ASSERT(maybe_chunk_path.status().ok());
+  chunk_path = maybe_chunk_path.value();
+  std::cout << "path of vertex property chunk for vertex id 520: " << chunk_path
+            << std::endl;
+  // next chunk
+  ASSERT(reader.next_chunk().ok());
+  maybe_chunk_path = reader.GetChunk();
+  ASSERT(maybe_chunk_path.status().ok());
+  chunk_path = maybe_chunk_path.value();
+  std::cout << "path of next vertex property chunk: " << chunk_path
+            << std::endl;
+  std::cout << "vertex property chunk number: " << reader.GetChunkNum()
+            << std::endl;
+}
+
+void adj_list_chunk_info_reader(const GAR_NAMESPACE::GraphInfo& graph_info) {
+  // constuct reader
+  std::string src_label = "person", edge_label = "knows", dst_label = "person";
+  auto maybe_reader = GAR_NAMESPACE::ConstructAdjListChunkInfoReader(
+      graph_info, src_label, edge_label, dst_label,
+      GAR_NAMESPACE::AdjListType::ordered_by_source);
+  ASSERT(maybe_reader.status().ok());
+  auto& reader = maybe_reader.value();
+
+  // use reader
+  auto maybe_chunk_path = reader.GetChunk();
+  ASSERT(maybe_chunk_path.status().ok());
+  auto chunk_path = maybe_chunk_path.value();
+  std::cout << "path of first adj_list chunk: " << chunk_path << std::endl;
+  // seek src
+  ASSERT(reader.seek_src(100).ok());
```

Comment:
Missing the `seek_dst`

Reply:
Maybe seek_src is enough here? seek_dst is invalid in this case.  I think we do not need to include all interfaces in a single example, it would be too complex.

Reply:
agree.